src/clr-angular/data/stack-view/stack-content-input.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Directive, Optional } from '@angular/core';
+import { StackBlockService } from './stack-block.service';
+
+@Directive({
+  selector: 'clr-stack-content > input[type="text"]',
+  host: {
+    '[class.clr-input]': 'true',
+    '[attr.aria-describedby]': 'stackBlockService?.id',
+  },
+})
+export class ClrStackContentInput {
+  constructor(@Optional() public stackBlockService: StackBlockService) {}
```

Comment:
Is there a path whereby `stackBlockService` would not be defined?

Reply:
The problem was that I forgot Angular doesn't support child combinator selectors, so `clr-stack-block > input[type="text"]` was just turning into `input[type="text"]` and matching too many elements. I fixed this by making the directive selector unique but apps will need to add this directive to their inputs to get the behaviors to work properly.